lib/ansible/plugins/lookup/hashi_vault.py

```diff
@@ -271,6 +300,54 @@ def auth_approle(self, **kwargs):
 
         self.client.auth_approle(role_id, secret_id)
 
+    def auth_aws_iam(self, **kwargs):
+        access_key = kwargs.get('access_key')
+        secret_key = kwargs.get('secret_key')
+        session_token = kwargs.get('session_token')
+
+        if access_key is None or secret_key is None:
+            import boto3
+            creds = boto3.Session().get_credentials()
+            if creds is None:
+                raise AnsibleError("Authentication method aws iam requires AWS credentials")
+            access_key, secret_key, session_token = creds.access_key, creds.secret_key, creds.token
+
+        header_value = kwargs.get('header_value')
+
+        mount_point = kwargs.get('mount_point')
+        if mount_point is None:
+            mount_point = 'aws'
+
+        role = kwargs.get('role')
+        if role is None:
+            raise AnsibleError("Authentication method aws iam requires a role")
+
+        region = kwargs.get('region', os.environ.get('AWS_DEFAULT_REGION'))
+        if region is None:
+            region = 'us-east-1'
+
+        self.client.auth_aws_iam(access_key, secret_key, session_token, header_value, mount_point, role, region=region)
```

Comment:
The `auth_aws_iam` method is deprecated:

> ansible/plugins/lookup/hashi_vault_iam_support.py:328: DeprecationWarning: Call to deprecated function 'auth_aws_iam'. This method will be removed in version '0.11.2' Please use the 'iam_login' method on the 'hvac.api.auth_methods.aws' class moving forward.

Use the `iam_login` as follows:

```suggestion
        self.client.auth.aws.iam_login(access_key, secret_key, session_token=session_token, header_value=header_value, mount_point=mount_point, role=role)
```

Note: I removed the `region=region` so authentication also works in non-default regions, see: https://hvac.readthedocs.io/en/stable/usage/auth_methods/aws.html#caveats-for-non-default-aws-regions

Reply:
Great thanks for the feedback!

I will fix the deprecated auth method shortly.

For the non-default region part, I have already handled the case by setting the `region` to `us-east-1` (at line 326-327) when `region` is not set in the lookup argument, which is the same value as the `iam_login ` default `region` value, so I will keep the optional region lookup argument. 
ref: https://github.com/hvac/hvac/blob/master/hvac/api/auth_methods/aws.py#L569